cobbler/items/item.py

```diff
@@ -467,39 +458,91 @@ def _resolve(self, property_name: str) -> Any:
+        ):
+            # This is only valid for profiles and is a workaround.
+            return getattr(getattr(self, "distro"), property_name)
+        if (
+            hasattr(self, "profile")
+            and getattr(self, "profile") != ""
+            and hasattr(
+                self.api.find_profile(name=getattr(self, "profile")), property_name
+            )
+        ):
+            # This is only valid for systems and is a workaround.
+            return getattr(
+                self.api.find_profile(name=getattr(self, "profile")), property_name
+            )
+        if (
+            hasattr(self, "image")
+            and getattr(self, "image") != ""
+            and hasattr(self.api.find_image(name=getattr(self, "image")), property_name)
+        ):
+            # This is only valid for systems and is a workaround.
+            return getattr(
+                self.api.find_image(name=getattr(self, "image")), property_name
+            )
+        if hasattr(settings, settings_name):
+            return getattr(settings, settings_name)
+        if hasattr(settings, f"default_{settings_name}"):
+            return getattr(settings, f"default_{settings_name}")
+        return None
+
+    def _resolve(self, property_name: str) -> Any:
+        """
+        Resolve the ``property_name`` value in the object tree. This function traverses the tree from the object to its
+        topmost parent and returns the first value that is not inherited. If the tree does not contain a value the
+        settings are consulted.
+
+        :param property_name: The property name to resolve.
+        :raises AttributeError: In case one of the objects try to inherit from a parent that does not have
+                                ``property_name``.
+        :return: The resolved value.
+        """
+        attribute_value, settings_name = self.__common_resolve(property_name)
 
         if attribute_value == enums.VALUE_INHERITED:
-            logical_parent = self.logical_parent
-            if logical_parent is not None and hasattr(logical_parent, property_name):
-                return getattr(logical_parent, property_name)
-            if hasattr(settings, settings_name):
-                return getattr(settings, settings_name)
-            if hasattr(settings, f"default_{settings_name}"):
-                return getattr(settings, f"default_{settings_name}")
-            AttributeError(
+            possible_return = self.__resolve_get_parent_or_settings(
+                property_name, settings_name
+            )
+            if possible_return is not None:
+                return possible_return
+            raise AttributeError(
                 f'{type(self)} "{self.name}" inherits property "{property_name}", but neither its parent nor'
-                f"settings have it"
+                f" settings have it"
             )
 
         return attribute_value
 
+    def _resolve_list(self, property_name: str) -> List[Any]:
```

Comment:
Why is this special case needed?

Reply:
Actually, you are right. Since now this is behaving like a scalar value we can remove this special case. This is a leftover from before another version of the bugfix. I will remove it.